dso_pkg/dso/task/regression/regression.py

```
@@ -107,9 +108,40 @@ def __init__(self, function_set, dataset, metric="inv_nrmse",
+                    output_cols = [col for col in df.columns if col.startswith(('y', 'Y'))]
+                    X_data = df[input_cols].values.astype(np.float64)
+                    y_data = df[output_cols].values.astype(np.float64)
+                    # If multiple outputs, keep as 2D; if single output, flatten
+                    if y_data.shape[1] == 1:
+                        y_data = y_data.flatten()
+                else:
+                    # No clear input/output structure, assume last column is output
+                    df = df.apply(pd.to_numeric, errors='coerce')
+                    X_data = df.values[:, :-1].astype(np.float64)
+                    y_data = df.values[:, -1].astype(np.float64)
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)